### PR TITLE
annotation for using TCP loopback for Envoy admin endpoint

### PR DESCRIPTION
**Pull request by yxue**
https://github.com/istio/istio/issues/19684

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -188,6 +188,13 @@ annotations:
     hidden: false
     resources:
       - Pod
+  - name: sidecar.istio.io/httpAdminEndpoint
```
Comment (ramaraochavali) [inline annotation/annotations.yaml]: IMO `httpAdminEndpoint` is little confusing and not providing enough context to figure out what this annotation will do. How about `adminEndpointOnUDS` or `secureAdminEndpoint` ?

Comment (yxue) [reply annotation/annotations.yaml]: `adminEndpointOnUDS` SGTM

**Comment by howardjohn** [comment]
I think this should be in mesh config

File: annotation/annotations.gen.go [hunk 2]
```diff
@@ -269,6 +269,16 @@ var (
 		  Resources: []ResourceTypes{ Pod, },
         }
 	
+		SidecarAdminEndpointOnUDS = Instance {
+          Name: "sidecar.istio.io/adminEndpointOnUDS",
+          Description: "Specifies whether or not using TCP loopback address as "+
+                        "Envoy admin endpoint. Unix domain socket will be used if "+
```
Comment (rlenglet) [inline annotation/annotations.gen.go]: I think this is the wrong logic. Envoy should be configured to use UDS unconditionally. This annotation should only enable a TCP proxy to it in pilot-agent.

Comment (yxue) [reply annotation/annotations.gen.go]: The intention is to make the UDS as an option and default to TCP loopback.

**Comment by duderino** [comment]
@yxue can you add an item to discuss this in a networking control plane wg meeting?  

This one is important but not urgent so you can wait until it works for your timezone ;)

**Comment by yxue** [comment]
> @yxue can you add an item to discuss this in a networking control plane wg meeting?
> 
> This one is important but not urgent so you can wait until it works for your timezone ;)

Sure. I am going to add the item to the WG meeting once I am back to the GMT-8 zone.

**Pull request closed without merging**